tests/auth_tests/test_forms.py

```diff
@@ -1371,3 +1402,21 @@ def test_html_autocomplete_attributes(self):
                 self.assertEqual(
                     form.fields[field_name].widget.attrs["autocomplete"], autocomplete
                 )
+
+    def test_unusable_password(self):
+        """Tests if deselecting usable password sets an unusable password"""
+        user = User.objects.get(username="testclient")
+        form = AdminPasswordChangeForm(
+            user,
+            {
+                "password1": "",
+                "password2": "test",
+            },
+        )
+
+        with mock.patch.object(User, "set_unusable_password") as set_unusable_password:
+            valid = form.is_valid()
+            form.save(commit=True)
+
+        self.assertTrue(valid)  # Valid despite password empty/mismatch
+        set_unusable_password.assert_called_once()
```

Comment:
Instead of mocking, can we just use the following:

```suggestion
        self.assertIs(user.has_usable_password(), False)
```